backend/src/game/routes.ts

```diff
@@ -73,11 +74,14 @@ function genericError() {
   return {error: 'Game data invalid'};
 }
 
-function validate(data: any): boolean {
+function validate(data: GameData): boolean {
   if (!data.mitigations) return false;
   if (!data.simulation || !data.simulation.length) return false;
 
   try {
+    const scenario = scenarios[data.scenarioName];
+    const lastSimData = last(data.simulation)!;
+    if (scenario.dates.endDate > lastSimData.date) return false;
```

Comment:
maybe even `!==`

Reply:
True, `<` doesn't validate anyways.

Reply:
But, sorry, it seems this is already contained in `isFinished`, pls double check.